### ISSUE TITLE
`/print` option shows command line arguments but not the actual configured options

### ISSUE THREAD

**Issue by keyboardDrummer** [boogie-org/boogie#475]
When Boogie is run through another tool, the `/print` option outputs Boogie file that at the top of the file show the command line options passed to that other tool, but not all the non-default Boogie options that Boogie was run with.

It would be great if Boogie would print the command that allows executing Boogie in the same way as in which it is currently executing.

**Comment by shazqadeer** [comment]
I did not understand this issue, perhaps because I did not try the experiment of running Boogie through another tool. When I run Boogie on the command line with `/print` option, I see the options I supplied on the command-line being printed in a comment at the top of the output file. This seemed good to me. Are you saying the problem happens only when Boogie is run via Dafny, for example?

**Comment by keyboardDrummer** [comment]
> I did not understand this issue, perhaps because I did not try the experiment of running Boogie through another tool. When I run Boogie on the command line with `/print` option, I see the options I supplied on the command-line being printed in a comment at the top of the output file. This seemed good to me. Are you saying the problem happens only when Boogie is run via Dafny, for example?

Yes this issue only occurs when Boogie is running programmatically.

**Comment by RustanLeino** [comment]
Currently, a typical Boogie client (like Dafny) sets the Boogie options by assigning to fields in a `Boogie.CommandLineOptions` object. For example,

``` c#
BCLO.TimeLimit = 10;
BCLO.VerifySnapshots = 3;
```

Any such assignment is unknown to the machinery that responds to `/print`.

Here are five possibilities:

0. Let `/print` output the value of every possible option, whether or not the client has set the option.

1. Let `/print` output the value of every option that, at the time of `/print`-ing, has a value that is different from Boogie's default for that option.

2. Change each `Boogie.CommandLineOptions` property setter to not just set a backing field but also store away, for the later benefit of `/print`, any option/value that was set.

3. Add a method

    ``` c#
    void SetBoogieOption(string optionName, string optionValue)
    ```

    that parses the given `optionName` and `optionValue`, sets the corresponding 
    `Boogie.CommandLineOptions` field, and also remembers these settings so that `/print` can reproduce them.
    A client would then use this method rather than setting the `Boogie.CommandLineOptions` fields directly.

4. <practicality=medium, cheesiness=high> Add a field

    ``` c#
    string OtherThingsIWantPrintToSay;
    ```

    in `Boogie.CommandLineOptions`. Then, a client can set this field to whatever and `/print` can just output this string 
    along with the other options.

Any thoughts on these?
